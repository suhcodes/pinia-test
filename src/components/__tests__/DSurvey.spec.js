import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Survey from '../DSurvey.vue'

const mockSurvey = {
  showQuestionNumbers: 'off',
  checkErrorsMode: 'onValueChanged',
  questionErrorLocation: 'bottom',
  elements: [
    {
      name: 'Name',
      title: 'Enter your name:',
      type: 'text',
      isRequired: true
    },
    {
      name: 'Email',
      title: 'Enter your email:',
      type: 'text',
      isRequired: true,
      validators: [{ type: 'email' }]
    }
  ]
}

describe('Tests Survey', () => {
  it('renders properly', async () => {
    const wrapper = mount(Survey, {
      props: {
        model: mockSurvey
      }
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.element).toMatchSnapshot()
  })
})
