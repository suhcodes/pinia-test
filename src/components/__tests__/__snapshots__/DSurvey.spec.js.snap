// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests Survey > renders properly 1`] = `
<surveycomponent
  model="[object Object]"
/>
`;
